django/db/backends/base/schema.py

```diff
@@ -547,9 +547,7 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
                 for fk_name in rel_fk_names:
                     self.execute(self._delete_constraint_sql(self.sql_delete_fk, new_rel.related_model, fk_name))
         # Removed an index? (no strict check, as multiple indexes are possible)
-        if (old_field.db_index and not new_field.db_index and
-                not old_field.unique and not
-                (not new_field.unique and old_field.unique)):
+        if old_field.db_index and not old_field.unique and (not new_field.db_index or new_field.unique):
```

Comment:
I have expanded the comment. Please let me know if you find the new version clearer. I was trying to strike a balance between over explaining the bool logic and keeping it concise and direct.

To give you a sense of my approach to this problem, I too found the original conditions confusing in intent. Instead of starting with the original, I started with a matrix of all combinations of `old_field.db_index`, `old_field.unique`, `new_field.db_index`, and `new_field.unique`. Then one by one, thought hard if the combination of values should result in the index change. Hence the Wolfram Alpha opportunity for simplified version. I did this for all `if`s changed then compared them to the original.